re/api/logging.go

```diff
@@ -0,0 +1,169 @@
+//
+// Copyright (c) 2019
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// +build !test
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/re"
+)
+
+var _ re.Service = (*loggingMiddleware)(nil)
+
+type loggingMiddleware struct {
+	logger log.Logger
+	svc    re.Service
+}
+
+// LoggingMiddleware adds logging facilities to the core service.
+func LoggingMiddleware(svc re.Service, logger log.Logger) re.Service {
+	return &loggingMiddleware{logger, svc}
+}
+
+func (lm *loggingMiddleware) Info(ctx context.Context) (info re.Info, err error) {
+	defer func(begin time.Time) {
+		message := fmt.Sprintf("Method info took %s to complete", time.Since(begin))
+		if err != nil {
+			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			return
+		}
+		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+	}(time.Now())
+
+	return lm.svc.Info(ctx)
+}
+
+func (lm *loggingMiddleware) CreateStream(ctx context.Context, token string, stream re.Stream, update bool) (result string, err error) {
+	method := "create_stream"
```

Comment:
why not a constant

Reply:
I do not separate update and create method, because they are quasi identical. There would be too much code repretion if i separate them.